src/Mvc/test/Mvc.IntegrationTests/ValidationIntegrationTests.cs

```diff
@@ -605,6 +605,78 @@ public async Task Validation_RequiredAttribute_OnPOCOPropertyOfBoundElement_NoDa
         AssertRequiredError("ProductId", error);
     }
 
+#nullable enable
+    private class ParameterInfos
+    {
+        public void Method(
+            string param1,
+            string param2 = "sample_data")
```

Comment:
@pranavkm Do you know of a good test class to validate that an action with this signature gets called with `param2 = "sample_data"` when `param2` cannot be bound from the request?

Reply:
To answer my own question, I think that a test similar to `Controller_WithDefaultParameterValue_ReturnsDefault` but using `[ApiController]` with nullability enabled should do the trick.

https://github.com/dotnet/aspnetcore/blob/76c3bec8a1e9ce265cf1e94b8d05fcb4150f7b16/src/Mvc/test/Mvc.FunctionalTests/DefaultValuesTest.cs#L48-L60

https://github.com/dotnet/aspnetcore/blob/76c3bec8a1e9ce265cf1e94b8d05fcb4150f7b16/src/Mvc/test/WebSites/BasicWebSite/Controllers/DefaultValuesController.cs#L18-L22

Reply:
@halter73, It should do the trick, however, what I did not understand (sorry if I am missing something) is which part of the code this test will cover, that is not covered by the integration test yet?

Reply:
@halter73, I added a new functional test to cover the end-to-end scenario here https://github.com/dotnet/aspnetcore/pull/38801/commits/20f69813b9985f568849f7a431ea092dc943e5dc